### DISCUSSION TITLE
Manually adding new bounding boxes based on the position of darknet predictions.

### DISCUSSION THREAD

**Discussion by orlandocloss** [AlexeyAB/darknet#8390] [Q&A] [answered]
Hi there,

I have used darknet in order to identify tomatoes as being Red, Green, Orange and Batch (Group of 8+ tomatoes).

However, while yolo is successful in identifying the individual tomatoes the Batch is not very accurate. I think what would produce a more accurate model is identifying when there is a cluster of tomatoes together and then drawing a new bounding box from the top left most x,y coordinate of all the identified tomatoes to the bottom right most x,y coordinate. 

Any suggestions on where to start in order to achieve this. Can I get a tabular data of the YOLO predictions and then edit this?

Would appreciate any help.

Many thanks,

Orlando

**Comment by stephanecharette** [comment, marked as answer]
Search for "Pseudo-labelling" in the [readme](https://github.com/AlexeyAB/darknet).  Is that what you want?

Your batch idea does work.  I use it in my "city" project, where my classes include "many vehicles":  https://www.youtube.com/watch?v=I-79ff1TD5M&t=24s

The other thing that may be of interest to you is the JSON output if you are trying to identify where you can go back and label some "batch" tomatoes:  https://www.ccoderun.ca/programming/darknet_faq/#json_output

**Reply by orlandocloss** [reply]
Thank you.
